src/sentry/integrations/aws_lambda/integration.py

```diff
@@ -62,6 +63,8 @@
     aspects={},
 )
 
+_lambda_setup_thread_pool = ThreadPoolExecutor(max_workers=10)
```

Comment:
Since this stuff is so rare, there's not much value in reusing globally and threads are rather cheap. I'd prefer we use the `with` style of execution when needed so the pool is created and torn down fully when finished. This way we don't keep a bunch of excess threads around that's rarely used.